modules/local/irma.nf

```diff
@@ -1,6 +1,6 @@
 process IRMA {
   tag "$meta.id"
-  label 'process_high'
```

Comment:
![image](https://github.com/user-attachments/assets/c9ca1dc9-d244-4ec9-ab38-eceea02c00c4)

Given that some IRMA processes are taking up to 8 hours, we can either keep it as `process_long` (20 hours) or increase `process_high` to at least 12 hours from its default value which is currently at 4 hours. However, I think using `process_long` in this context is the better option as it accounts for longer times if they ever come up.

Let me know which you prefer and I'll make the change @peterk87

Reply:
That's a huge issue that IRMA is taking so long and that the runtime seems to be scaling exponentially with the size of the input. I recently came into this issue with IRMA on a metagenomics dataset of ~1.5GB fastq.gz. Although there were only a handful of Influenza reads in the dataset, IRMA spent a huge amount of time going through every single read individually rather than just getting to the reads of interest. The amount of IO was ridiculous with at least 10X the size of the initial dataset uncompressed written and rewritten over and over. The kicker is that the analysis failed due to some random issue conversion from some custom format to FASTA/FASTQ format.

We can look into updating IRMA to the latest version. Initial tests found some striking differences in results, but those might have been fixed in a later release (see [release notes](https://wonder.cdc.gov/amd/flu/irma/release_notes.html)). For some reason, the latest version on [Docker Hub](https://hub.docker.com/r/cdcgov/irma/tags) is 1.2.0 while on their site it's 1.1.4! The Bioconda version is at [1.0.3](https://github.com/bioconda/bioconda-recipes/blob/9a91d82caeca71daf84d6e9755458beae86b6940/recipes/irma/meta.yaml#L2).

At this point, I'm wondering if it'd be faster to just map all reads against all public and user provided Influenza sequences using Minimap2, and/or de novo assemble with Megahit if working with Illumina data, and perform BLASTN to determine the top refs. Mash screen or some kind of kmer profile scanning may also be a possibility to skip having to ref or de novo assemble and run BLAST.

Reply:
I've actually been quite interested in using Mash screen as an alternative to IRMA.  IRMA is really cool, but yeah, it is routinely very slow.  In the end, the only output I consistently look at is the `READ_COUNTS.txt` file in `irma --> tables` to help suss out whether a sample is mixed/contaminated.  In any case, I am personally not opposed to any route that gets us to the same spot, but I do recognize that this will be a more significant refactor and want to be respectful of yours and Abdallah's time.

Reply:
Given the current PR is focused on the Illumina expansion, and IRMA is native to both Nanopore and Illumina workflows, I suggest we tackle this refactor in a separate PR as to push the current changes out. That will allow time for a full refactor of the IRMA process if needed.

Reply:
Agreed! I've created some issues related to IRMA (#81 and #80) that can be addressed in future PRs.